### PR TITLE
Feature/circleci and docker

### PR DESCRIPTION
**Pull request by olevs**
Initial CircleCI setup with custom Docker image for the ALive project

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
+version: 2.1
+jobs:
+  build:
+    working_directory: /usr/src/Slicer-LiverAnalysis
+    docker:
+      - image: olevs/slicer-alive-build
```
Comment (RafaelPalomar) [inline .circleci/config.yml]: We should use [slicer/slicer-build](https://hub.docker.com/r/slicer/slicer-build). It has all we need and it is maintained by the Slicer devs

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    steps:
+      - checkout
+      - run:
+          name: Update 3D Slicer
+          command: |
+            cd /usr/src/Slicer
+            git checkout .
+            git pull
```
Comment (RafaelPalomar) [inline .circleci/config.yml]: If we use the [slicer/slicer-build](https://hub.docker.com/r/slicer/slicer-build) image, here we would do the git clone, etc..

File: .circleci/config.yml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+      #    name: Update LiverAnalysis plugin
+      #    command: |
+      #      cd /usr/src/Slicer-LiverAnalysis
+      #      git checkout .
+      #      git pull
+      
+      - run:
+          name: Build LiverAnalysis plugin
+          environment:
+            BUILD_TOOL_FLAGS: "-j3"
```
Comment (RafaelPalomar) [inline .circleci/config.yml]: I would leave this as `-j1` just to start. I think extensions will not take much to compile and we will probably get only one core with the free plan anyways

File: .circleci/config.yml [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+      # Not needed, as this is done by the checkout step
+      #- run:
+      #    name: Update LiverAnalysis plugin
+      #    command: |
+      #      cd /usr/src/Slicer-LiverAnalysis
+      #      git checkout .
+      #      git pull
+      
+      - run:
+          name: Build LiverAnalysis plugin
```
Comment (RafaelPalomar) [inline .circleci/config.yml]: plugin -> extension. In the slicer jargon there are no plugins, but  extensions or modules. Let's try to keep their language (sorry for the pickiness :)

**Review by RafaelPalomar — CHANGES_REQUESTED** [review]
Thanks for working on this, @olevs! 

Long story short: I think we should simplify the CI system to only use the [slicer/slicer-build](https://hub.docker.com/r/slicer/slicer-build) docker image as a base. Thus, we don't have to maintain any docker images. Note that we don't wan to diverge from the official Slicer (therefore no need to use our fork of Slicer).

File: .circleci/config.yml (hunk 1 above)
Comment (olevs) [reply .circleci/config.yml]: slicer/slicer-build don't work, and isn't maintained. It's slicer/slicer-base that is the image to use.
Using this image will add +1,5 hours to each build as the build files are deleted. It don't seem like Slicer keeps any docker images with build files. That's the main reason for creating a new image.

Comment (olevs) [reply .circleci/config.yml]: To get something up and running I used our fork of Slicer, as the latest changes in the main fork haven't compiled for the last week. I just tried again to compile it inside slicer/slicer-build, and it still fails.

Comment (RafaelPalomar) [reply .circleci/config.yml]: @olevs  you are right, the slicer-build image looks not maintained. Then I suggest the following:

1. Keep all the needed docker image definitions for ALive in a single repository ([ALive-Docker](https://github.com/ALive-research/ALive-Docker)). For now, we can create our own `slicer-build` based on `SlicerDocker/slicer-base`. I would remove  the `slicer-alive-build` in favor of checkout/builld/test commands in the circleci configuration. It is easier to maintain.
2. Have the container `slicer-build` image in [aliveresearch/slicer-build](https://hub.docker.com/repository/docker/aliveresearch/slicer-build) dockerhub repo.

Comment (olevs) [reply .circleci/config.yml]: I agree. This sounds like a good solution. All the ALive stuff should be lightweight anyway, and don't need to be present in the Docker container.

File: .circleci/config.yml (hunk 2 above)
Comment (olevs) [reply .circleci/config.yml]: Using aliveresearch/slicer-build seems to work

File: .circleci/config.yml (hunk 3 above)
Comment (olevs) [reply .circleci/config.yml]: I set it to -j1 for now. It probably don't matter much. Slicer uses -j3.

File: .circleci/config.yml (hunk 4 above)
Comment (olevs) [reply .circleci/config.yml]: Changed

**Comment by RafaelPalomar** [comment]
We will need to add a testing section in the CircleCI configuration. But for now, since we don't have tests, we can approve this.

**Comment by RafaelPalomar** [comment]
Great work @olevs!

**Pull request merged**